modules/mono/glue/GodotSharp/GodotSharp/Core/Array.cs

```diff
@@ -655,26 +1187,103 @@ public void Shuffle()
         }
 
         /// <summary>
-        /// Searches this <see cref="Array{T}"/> for an item
-        /// and returns its index or -1 if not found.
+        /// Searches the array for a value and returns its index or <c>-1</c> if not found.
         /// </summary>
-        /// <param name="item">The item to search for.</param>
+        /// <param name="item">The <see cref="Variant"/> item to search for.</param>
         /// <returns>The index of the item, or -1 if not found.</returns>
         public int IndexOf(T item)
         {
+            if (Count == 0)
+            {
+                // Special case for empty array to avoid an interop call.
+                return -1;
+            }
```

Comment:
I'm assuming that accessing `Count` is much cheaper than invoking the interop call to `index_of`, so if the array is empty we can directly return `-1`. Let me know if my assumption is wrong.

Reply:
This optimization is missing from these methods on the non-generic Array.